api/transport/header.go

```diff
@@ -65,16 +67,23 @@ func (h Headers) With(k, v string) Headers {
 	if h.items == nil {
 		h.items = make(map[string]string)
 	}
+	if h.keyMapping == nil {
```

Comment:
These if blocks can be coalesced because `(keyMapping == nil) == (items == nil)` is invariant.